File: blood-bank-be/bloodBank/src/main/java/ftn/uns/ac/rs/bloodbank/blood/model/BloodContract.java
package ftn.uns.ac.rs.bloodbank.blood.model;
import lombok.*;

import javax.persistence.*;
import java.io.Serializable;
import java.util.Date;
import java.util.List;

@Entity(name = "BloodContract")
@Table(name = "blood_contract")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class BloodContract implements Serializable {
    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE)
    @Column(name = "id", nullable = false)
    private Long id;
    @ManyToMany
    private List<BloodUnit> bloodUnits;
    private Integer price;
    private Date deliveryDate;
    private Boolean isExpired = false;
    private String hospitalName;
}

File: blood-bank-fe/src/app/model/BloodContract.ts
import {BloodUnit} from "./BloodUnit";

export class BloodContract{
  public id?:string;
  public bloodUnits?:BloodUnit[];
  public dateOfDelivery?:Date;
  public bloodBankName?:String;
  public hospitalName?:String;
  public price?:number;
}
